### ISSUE TITLE
[FEATURE] Add arguments to build_image command

### ISSUE THREAD

**Issue by hawksight** [CircleCI-Public/gcp-gcr-orb#25]
### Orb version

0.5.1
Also not present in latest: 0.6.0

### What happened

Using the `build-image` [command](https://github.com/CircleCI-Public/gcp-gcr-orb/blob/master/src/commands/build-image.yml) your currently cannot pass any arguments to the docker build command. 

This is important for passing metadata into the container image such as the git commit SHA. There's probably a huge list of meta people might want to add.

### Expected behavior

It would be really useful to expose an argument list, or map and have those passed to the docker build command. The solution would allow people to specify the key values they want to embed within the image.

It might be something like:

```yaml
# command
      - gcp-gcr/build-image:
          registry-url: <<parameters.registry-url>>
          google-project-id: <<parameters.google-project-id>>
          image: <<parameters.image>>
          tag: << parameters.tag >>
          path-to-dockerfile: <<parameters.path-to-dockerfile>>
          arguments: <<parameters.build-arguments>>

# Using the job
      - build_and_push_image:
          image: $CIRCLE_PROJECT_REPONAME
          tag: $CIRCLE_SHA1
          build-arguments: 
            - GITHASH: $CIRCLE_SHA1
            - A_KEY: "arandomvalue"
          registry-url: gcr.io
          context: gcr-docker-images
          requires:
            - build
          filters:
            branches:
              only:
                - staging
                - prod
```

**Comment by iynere** [comment]
this is a good request. worth pointing out, you can do this w/ the docker orb:

https://github.com/CircleCI-Public/docker-orb/blob/master/src/commands/build.yml

**Comment by iynere** [comment]
maybe the gcp-gcr orb should just call the docker orb 🤔

**Comment by iynere** [comment]
we don't support a list or map type parameter, but we can allow a multi-line string @hawksight

**Comment by hawksight** [comment]
Thanks @iynere - in the spirit of orbs, I'd probably say that the build step should reuse the docker orb which has this feature. 

It's a little hacky having to pass it all in a string, but it should do what is needed. Sorry I hadn't look up the types, I had just assumed there would be lists and maps but I was wrong. So a string is the best option right now. I might give it a try locally for now.

**Comment by iynere** [comment]
patched in `circleci/gcp-gcr@0.6.1`
